src/test/kotlin/edu/carroll/cs389/service/tracker/trackservice/TrackServiceQueryTest.kt

```diff
@@ -0,0 +1,561 @@
+package edu.carroll.cs389.service.tracker.trackservice
+
+import edu.carroll.cs389.service.tracker.TrackService
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.util.AssertionErrors.assertTrue
+import java.sql.Timestamp
+import java.util.Date
+
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class TrackServiceQueryTest {
+
+    @Autowired
+    private lateinit var trackService: TrackService
+
+    /*******************************************************/
+    /*******************************************************/
+    /* THESE TESTS ALL PERTAIN TO THE queryDatabase METHOD */
+    /*******************************************************/
+    /*******************************************************/
+
+    /*************************************/
+    /* Happy Path Tests (expected input) */
+    /*************************************/
+
+    @Test
+    fun validUserTest() {
+        val ipv4: String = "192.168.1.1"
+        val browser: String = "Firefox"
+        val browserMajorVersion: String = "110"
+        val platform: String = "Ubuntu"
+        val platformVersion: String = "22.04 LTS"
+        val requestURI: String = "/"
+
+        // logging time to be between
+        val timeU1i = Timestamp(Date().time)
+        Thread.sleep(100)
```

Comment:
This could probably use a much lower value (`sleep(1)`) as I believe almost all storage services have at LEAST millisecond accuracy, allowing your tests to run slightly faster.